docs/Actions.md

```diff
@@ -9,57 +9,75 @@ React-admin provides special hooks to emit read and write queries to the [`dataP
 
-React-admin stores the `dataProvider` object in a React context, so it's available from anywhere in your application code. The `useDataProvider` hook grabs the Data Provider from that context, so you can call it directly.
+React-admin stores the `dataProvider` object in a React context, so it's available from anywhere in your application code. The `useDataProvider` hook grabs the Data Provider from that context, so you can call it directly. 
 
-For instance, here is how to query the Data Provider for the current user profile:
+As `dataProvider` methods are asynchronous, it's common to call them in a React `useEffect` (for queries) or in an event handler (for mutations).
+
+For instance, here is how to query the Data Provider for a User record on mount, combining the `useDataProvider` hook with the `useState` and `useEffect` hooks:
 
 ```jsx
+import { useState, useEffect } from 'react';
 import { useDataProvider } from 'react-admin';
+import { Loading, Error } from './MyComponents';
 
-const MyComponent = () => {
+const UserProfile = ({ userId }) => {
     const dataProvider = useDataProvider();
-    // ...
-}
-```
-
-Refer to [the `useDataProvider` hook documentation](./useDataProvider.md) for more information.
+    const [user, setUser] = useState();
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState();
+    useEffect(() => {
+        dataProvider.getOne('users', { id: userId })
+            .then(({ data }) => {
+                setUser(data);
+                setLoading(false);
+            })
+            .catch(error => {
+                setError(error);
+                setLoading(false);
+            })
+    }, []);
 
-**Tip**: The `dataProvider` returned by the hook is actually a *wrapper* around your Data Provider. This wrapper logs the user out if the `dataProvider` returns an error, and if the `authProvider` sees that error as an authentication error (via `authProvider.checkError()`).
+    if (loading) return <Loading />;
+    if (error) return <Error />;
+    if (!user) return null;
 
-## DataProvider Method Hooks
+    return (
+        <ul>
+            <li>Name: {user.name}</li>
+            <li>Email: {user.email}</li>
+        </ul>
+    )
+};
+```
 
-React-admin provides one hook for each of the Data Provider methods. They are useful shortcuts that make your code more readable and more robust.
+Refer to [the `useDataProvider` hook documentation](./useDataProvider.md) for more information.
 
-The query hooks execute on mount. They return an object with the following properties: `{ data, isPending, error }`. Query hooks are:
+**Tip**: For standard data provider methods (like  `getOne()` or `update()`), you should use the [Query hooks](#dquery-hooks) and the [Mutation hooks](#mutation-hooks), both documented below, instead of calling `useDataProvider`. These hooks are easier to use, more robust, they handle the loading state for you, and they are typed.
```

Comment:
```suggestion
**Tip**: For standard data provider methods (like  `getOne()` or `update()`), you should use the [Query hooks](#query-hooks) and the [Mutation hooks](#mutation-hooks), both documented below, instead of calling `useDataProvider`. These hooks are easier to use, more robust, they handle the loading state for you, and they are typed.
```